js/DataService.js

```diff
@@ -42,9 +41,19 @@ angular.module('PathOfDamage')
           description: "Modifiers that shift physical damage to elemental, typically reading like '% of Physical Damage taken as Y', such as Taste of Hate or Lightning Coil",
           tables: {
             shifts: {
-              totalName: "Damage Shifted",
-              totalMin: 0,
-              totalMax: 100
+              name: "Damage Shifted",
+              calcTotal: function () {
+               var totals = {};
+               var total = 0;
+                this.values.forEach(function (shift) {
+                  if (shift.enabled && shift.value && shift.element) {
+                    totals[shift.element] = totals[shift.element] + shift.value || shift.value;
+                    total += shift.value;
+                  }
+                });
+                this.totals = totals;
```

Comment:
These variable names are confusing. The 'total' is not in the set of totals, for example. The total object should reference elemental or something, and maybe displayTotal is a better name for the calculated result shown at the bottom of the tables.